### PR TITLE
Setup association

### PR DESCRIPTION
**Pull request by Basir-Mohammadi**
## Project requirements

- Create a class Classroom with the following:
- @label instance variable, should be initialized in the constructor.
- Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).
- Make sure that when adding a student to a classroom it also sets the classroom for the student.
- Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
- @title and @author instance variables, should be initialized in the constructor.
- Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
- @date instance variable, should be initialized in the constructor.
- Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- Create the has-many side of Book and Rental (a book has many rentals).
- Create the belongs-to side of Rental and Book (a rental belongs to a book).
- Create the has-many side of Person and Rental (a person has many rentals).
- Create the belongs-to side of Rental and Person (a rental belongs to a person).
- Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  def initialize(title:, author:)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person)
+    Rental.new(person: person, book: self)
+  end
```
Comment (danmainah) [inline book.rb]: - [x] Kindly add a date parameter when creating your add_rental method, you will use the date as a parameter to create a new rental.

File: person.rb [hunk 2]
```diff
@@ -10,6 +11,11 @@ def initialize(age:, name: 'unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(book)
+    Rental.new(person: self, book: book)
   end
```
Comment (danmainah) [inline person.rb]: - [ ] Kindly add a date parameter when creating your add_rental method, you will use the date as a parameter to create a new rental.

File: person.rb [hunk 3]
```diff
@@ -1,7 +1,8 @@
 require './nameable'
+require './rental'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (danmainah) [inline person.rb]: - [x] Kindy move the rentals to `attr_accessor, the rentals should have both Getters and Setters.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Basir-Mohammadi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by mwanawabangona — APPROVED** [review]
Hi @Basir-Mohammadi,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**